packages/governance/docs/AttackGuide.md

```diff
@@ -0,0 +1,101 @@
+
+I see two main kinds of threats here:
+
+ * bugs or oversights in the infrastructure
+ * attacks on voters
+ * mistakes that might arise in governed contracts 
+
+The first two can be mitigated by validation functions that cross-check linkages.
+
+I'm not as sure how to address the third, except better documentation, and support for custom validation functions.
+
+
+# Get a voter facet you shouldn't have
+
+Every module that handles voter facets should handle them carefully and ensure they don't escape. If you 
+
+# Get a ballotCounter to accept votes from a single voter without replacement
+
+I can't think of a way to evade the way BinaryBallotCounter handles this, but it's something to be aware of with new BallotCounters, and particularly in combination with new Registrars.
+
+# Break notification so voters don't hear about elections
+
+This would be a minor issue, but the notifications are the only mechanism we proved for announcing new issues.
+
+# Leak the ability to create new questions
+
+Currently creating new questions is seen as tightly held. If that is loosened in a new Registrar or ElectionManager, that provides a path for spamming voters.
+
+# What shenanigans can be caused by creating multiple questions with the same or very similar text?
+
+The question text used to be unique, but each ballot question now has a handle for disambiguation. Voters ought to validate the particulars of any question they intend to vote on. Can they be confused by corrections or replacement? Is there a vulnerability here, or just a UI support need?
+
+# Create a Ballot that refers to a different BallotCounter than the one the registrar will use
+# (Attacker) distribute ballots that don't match the official one from the Registrar
+
+Ballots themselves are not secure. The voter has to get a copy of the ballot from the Registrar to have any assurance that it's valid. If someone else provides a ballot, they can replace various pieces to fool the voter as to what is being voted on.
+
+# Ordinary bugs in counting ballots, reporting, etc.
+
+If the code in BallotCounter, Registrar, ContractGovernor has subtle mistakes, wrong results will obtain.
+
+# Produce a discrepancy between Terms and actions in BallotCounter or Registrar
+
+The voter's assurance that a particular ballot has the effect they expect arises in part because the `terms` in the BallotCounter, Registrar, etc. dictate how those classes will act. If the code is changed to get info from hidden parameters or to ignore some of the terms, voters will be misled.
+
+# Use a timer that is controlled by a party to the vote
+
+Everyone involved relies on the timers used to close voting being platform timers, but the timers aren't self-revealing. Participants should compare the timers to known platform-provided timers before relying on them.
```

Comment:
Link to https://github.com/Agoric/agoric-sdk/issues/3748 ?

Reply:
done